collect_app/src/main/java/org/odk/collect/android/google/DriveHelper.java

```diff
@@ -0,0 +1,216 @@
+
+    DriveHelper(GoogleAccountCredential credential, HttpTransport transport, JsonFactory jsonFactory) {
+        drive = new Drive.Builder(transport, jsonFactory, credential)
+                .setApplicationName("ODK-Collect")
+                .build();
+    }
+
+    /**
+     * Returns id of the root folder or null
+     */
+    public String getRootFolderId() throws IOException {
+        return drive.files()
+                .get("root")
+                .setFields("id")
+                .execute().getId();
+    }
+
+    public Drive.Files.List buildRequest(String query) {
+        try {
+            return drive.files()
+                    .list()
+                    .setQ(query)
+                    .setFields("nextPageToken, files(modifiedTime, id, name, mimeType)");
+
+        } catch (IOException e) {
+            Timber.e(e);
+        }
+        return null;
+    }
+
+    public boolean downloadFile(DriveListItem fileItem, HashMap<String, Object> results) {
+        FileOutputStream fileOutputStream = null;
+        try {
+            fileOutputStream = new FileOutputStream(new File(Collect.FORMS_PATH + File.separator + fileItem.getName()));
+            downloadFile(fileItem.getDriveId()).writeTo(fileOutputStream);
+        } catch (IOException e) {
+            Timber.e(e);
+            results.put(fileItem.getName(), e.getMessage());
+            return false;
+        } finally {
+            try {
+                if (fileOutputStream != null) {
+                    fileOutputStream.close();
+                }
+            } catch (IOException e) {
+                Timber.e(e, "Unable to close the file output stream");
+            }
+        }
+        return true;
+    }
+
+    private ByteArrayOutputStream downloadFile(String fileId) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        drive.files().get(fileId).executeMediaAndDownloadTo(outputStream);
+        return outputStream;
+    }
+
+    public String createOrGetIDOfFolderWithName(String jrFormId)
+            throws IOException, MultipleFoldersFoundException {
+        String submissionsFolderId = createOrGetIDOfSubmissionsFolder();
+        return getIDOfFolderWithName(jrFormId, submissionsFolderId);
+    }
+
+    private String createOrGetIDOfSubmissionsFolder()
+            throws IOException, MultipleFoldersFoundException {
+        String rootFolderId = getIDOfFolderWithName(GOOGLE_DRIVE_ROOT_FOLDER, null);
+        return getIDOfFolderWithName(GOOGLE_DRIVE_SUBFOLDER, rootFolderId);
+    }
+
+    public String getIDOfFolderWithName(String name, String inFolder) throws IOException, MultipleFoldersFoundException {
+
+        com.google.api.services.drive.model.File folder = null;
+
+        // check if the folder exists
+        ArrayList<com.google.api.services.drive.model.File> files = getFilesFromDrive(name, inFolder);
+
+        for (com.google.api.services.drive.model.File file : files) {
```

Comment:
This is a confusing way to do this: can we just check if `files.size() > 1` instead?